test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4247,45 +4250,85 @@ def _load_configuration(self, infile, objtype=None):
-                        fd.write("\n".join(v))
-                        self.du.run_cmd(self.hostname, [qmgr], cstdin=fd,
-                                        sudo=True)
-                    else:
-                        fd.write("\n".join(v))
-                        # append the last line
-                        fd.write("\n")
-                        self.du.run_cmd(self.hostname, ['cp', fn, k],
-                                        sudo=True)
-                    os.remove(fn)
-
+                # Load server configuration
+                if k.startswith('qmgr_'):
+                    fd, path = tempfile.mkstemp(prefix="runcmd_")
+                    os.chmod(path, 0o700)
+                    with open(path, 'a') as f:
+                        f.write('\n'.join(v))
+                    qmgr_cmd = qmgr + " < " + path
+                    d = self.du.run_cmd(
+                        self.hostname, qmgr_cmd, sudo=True, as_script=True)
+                    os.unlink(path)
+                # Load pbs.conf file
+                elif k == "pbs_conf":
+                    cfg_vals = base64.decodestring(v)
+                    conf_path = os.path.join(os.sep, 'etc', 'pbs.conf')
+                    # Clear current pbs.conf file and
+                    # rewite with saved config contents
+                    with open(conf_path, 'w') as f:
+                        f.truncate(0)
+                        f.write(cfg_vals)
+                # Load hooks
+                elif k == "hooks":
+                    for hpath in conf['hooks']:
+                        h_name = hpath.split(os.sep)[-1]
+                        hook_name = h_name[:-3]
+                        hook_name = str(hook_name)
+                        # Skip loading PBS hook
+                        if hook_name.startswith("PBS_"):
+                            continue
+                        cf_hpath = ""
+                        if ".PY" in hpath:
+                            cf_hpath = hpath.replace('.PY', '.CF', 1)
+                            hk_hpath = hpath.replace('.PY', '.HK', 1)
+                            rep_str = "hook_name=" + hook_name
+                            cmd = qmgr + " -c 'create hook %s'" % hook_name
+                            self.du.run_cmd(
+                                self.hostname, cmd, sudo=True,
+                                as_script=True)
+                            hk_contents = base64.decodestring(
+                                conf['hooks'][hk_hpath])
+                            hook_py_b64str = conf['hooks'][hpath]
+                            hook_py_str = base64.decodestring(
+                                hook_py_b64str)
+                            hk_vals = self.utils.convert_to_dictlist(
+                                hk_contents.splitlines())
+                            for k, val in hk_vals[0].items():
+                                if k == 'event':
+                                    events = "'%s'" % val
```

Comment:
Where is the 'events' variable being used?
Also, the function Sujata is referring to is "create_import_hook".

Reply:
Removed variable events